.circleci/config/jobs/website-docker-image.yml

```diff
@@ -0,0 +1,22 @@
+docker:
+  - image: circleci/buildpack-deps
+shell: /usr/bin/env bash -euo pipefail -c
+steps:
+  - checkout
+  - setup_remote_docker
+  - run:
+      name: Diff package-lock.json
+      command: |
+        echo 'export PACKAGE_LOCK_CHANGED=$(git diff --name-only $(git log --pretty=format:'%h' -n1 HEAD~1)...HEAD | grep -c website/package-lock.json)' >> $BASH_ENV
+  - run:
+      name: Build Docker Image if Necessary
+      command: |
+        if [ "$CIRCLE_BRANCH" = "master" ] && [ $PACKAGE_LOCK_CHANGED -gt 0 ]; then
```

Comment:
Can you switch this to using a workflow filter for `stable-website` again?

Reply:
This task detects when dependencies have changes and deploys a new docker image bundled with the latest dependencies. The goal here is that once a PR is merged to master that changes the dependencies in any way, everyone else working on the site gets the most up to date image. In this case we are purposely targeting `master`, since pull requests are typically made to master and cherry-picked into stable website, at least as far as I know.

Reply:
@jescalan this is generally true - although I've definitely seen doc hotfixes go directly to stable-website in the past.

Selfishly mostly wanted us to avoid having _another_ CI check for every branch bc they get pretty noisy. Could also follow a convention similar to UI branches where it will run on `master` and `doc-*` branches?

Reply:
> I've definitely seen doc hotfixes go directly to stable-website in the past.

Is there any scenario where a doc hotfix would go to stable-website and not touch master at all? If so, we'd need to create some sort of special case for that, but it feels like a super dangerous practice to me.

> mostly wanted us to avoid having another CI check for every branch bc they get pretty noisy

The line above that we're commenting on specifies that this task runs only when dependencies on the website change (`package-lock.json` touched) and only on the `master` branch, otherwise it just exits and doesn't run anything. If there's a different way that this could be implemented though I'm very much open to that!

Reply:
> Is there any scenario where a doc hotfix would go to stable-website and not touch master at all? If so, we'd need to create some sort of special case for that, but it feels like a super dangerous practice to me.

I'm fine with forcing every docs change to go through `master` before going into `stable-website`.